web/src/infra/common/components/base/FreeBoardGamesBar.tsx

```diff
@@ -1,55 +1,102 @@
+/* eslint-disable react/prop-types */
 import React from 'react';
+import Select from '@material-ui/core/Select';
+import MenuItem from '@material-ui/core/MenuItem';
 import AppBar from '@material-ui/core/AppBar';
 import Toolbar from '@material-ui/core/Toolbar';
 import Typography from '@material-ui/core/Typography';
 import FbgLogo from './media/fbg_logo_white_48.png';
 import { Link } from 'infra/i18n';
 import { home } from 'infra/navigation';
+import { makeStyles } from '@material-ui/core';
+import { nextI18Next } from 'infra/i18n/config';
+import LanguageIcon from '@material-ui/icons/Language';
 
-interface FBGBarProps {
+type FBGBarProps = {
   FEATURE_FLAG_readyForDesktopView?: boolean;
   toolbarContent?: React.ReactNode;
-}
+};
+
+const handleLanguageChange = (event) => {
+  window.location.href = `/${event.target.value}`;
+};
 
-class FreeBoardGamesBar extends React.Component<FBGBarProps, {}> {
-  render() {
-    const maxWidth = this.props.FEATURE_FLAG_readyForDesktopView ? '1200px' : '500px';
+function renderValue() {
+  return <LanguageIcon />;
+}
 
-    return (
-      <>
-        <div
-          style={{
-            marginLeft: 'auto',
-            marginRight: 'auto',
-          }}
-        >
-          <AppBar position="sticky">
-            <Toolbar>
-              <Link href={() => home()}>
-                <a style={{ textDecoration: 'none', display: 'flex', alignItems: 'center' }}>
-                  <img style={{ marginRight: '8px', height: '48px' }} src={FbgLogo} alt="FbG" />
-                  <Typography component="h1" variant="h6" style={{ color: 'white' }}>
-                    FreeBoardGames.org
-                  </Typography>
-                </a>
-              </Link>
-              {this.props.toolbarContent}
-            </Toolbar>
-          </AppBar>
-        </div>
-        <div
-          style={{
-            maxWidth,
-            marginLeft: 'auto',
-            marginRight: 'auto',
-          }}
-          data-testid={'childrenDiv'}
-        >
-          {this.props.children}
-        </div>
-      </>
-    );
-  }
+function renderLanguageSelect() {
+  const { i18n } = nextI18Next.useTranslation();
+  const color = 'white!important' as 'white';
+  const useWhite = makeStyles({
+    select: {
+      color,
+      '&:before, &:after': {
+        display: 'none!important' as 'none',
+      },
+    },
+    icon: {
+      fill: color,
+    },
+  });
+  const classes = useWhite();
+  return (
+    <div style={{ marginLeft: 'auto' }}>
+      <Select
+        value={i18n.language}
+        className={classes.select}
+        inputProps={{
+          classes: {
+            icon: classes.icon,
+          },
+        }}
+        onChange={handleLanguageChange}
+        renderValue={renderValue}
```

Comment:
Is there a reason you chose to use the `renderValue` variable, as opposed to keeping it inline?  Something like:

```typescript
renderValue={<LanguageIcon/>}
```

Reply:
Done. I was playing with more complex variations.